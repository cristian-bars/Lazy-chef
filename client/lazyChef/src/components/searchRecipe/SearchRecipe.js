/* eslint-disable no-shadow */
import React, {useEffect, useState} from 'react';
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  TextInput,
  FlatList,
} from 'react-native';
import {connect} from 'react-redux';
import {PropTypes} from 'prop-types';
import {useNavigation} from '@react-navigation/native';
import generalStyles from '../../../generalStyles';
import styles from './searchRecipeStyles';

const SearchRecipe = ({recipes, dispatch, route}) => {
  const navigation = useNavigation();
  const {ingredient} = route.params;
  let [userWord, searchIngredient] = useState('');
  let [userPalabra, setUserPalabra] = useState([ingredient]);
  let array = [];
  let [arrayToMap, setArrayToMap] = useState([]);
  //const [text, setText] = useState('');
  const addIngredient = () => {
    array.push(...userPalabra, userWord);
    setUserPalabra([...userPalabra, userWord]);
    setArrayToMap(array);
    //setText();
    console.log(arrayToMap);
  };

  const listRender = ({item}) => {
    return <Text>{item}</Text>;
  };

  const Words = () => {
    return (
      <View>
        <View style={styles.wordsList}>
          <Text style={styles.wordsTitle}>Mi selecci√≥n</Text>
          {arrayToMap.length ? (
            <FlatList
              style={styles.list}
              data={arrayToMap}
              keyExtractor={item => item}
              renderItem={listRender}
            />
          ) : (
            <Text>{ingredient}</Text>
          )}
          <Text style={styles.searchWord} />
        </View>
      </View>
    );
  };
  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <View style={styles.input}>
          <TouchableOpacity onPress={addIngredient}>
            <Image
              style={styles.searchImage}
              source={require('../../img/search.png')}
            />
          </TouchableOpacity>
          <TextInput
            style={styles.inputText}
            placeholder="Buscar"
            autoCapitalize="none"
            //value={text}
            onChangeText={ing => searchIngredient(ing)}
          />
        </View>
      </View>
      <View>
        <Words />
      </View>
      <TouchableOpacity
        style={generalStyles.button}
        onPress={() => {
          arrayToMap.length
            ? navigation.navigate('RecipeList', {ingredient: arrayToMap})
            : navigation.navigate('RecipeList', {ingredient: ingredient});
        }}>
        <Text style={generalStyles.baseText}>Mostrar recetas</Text>
      </TouchableOpacity>
    </View>
  );
};

SearchRecipe.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    recipes: store.recipes,
  };
}

export default connect(mapStateToProps)(SearchRecipe);
