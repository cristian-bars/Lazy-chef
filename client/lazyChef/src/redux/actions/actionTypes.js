export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
