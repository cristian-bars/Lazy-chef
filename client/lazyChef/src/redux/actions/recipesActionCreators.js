import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/recipes';
const url = 'http://192.168.0.29:2022/recipes';
//const url = 'http://192.168.1.130:2022/recipes';

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return {
    type: actionTypes.ADD_RECIPE,
    recipe,
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    await axios.delete(`${url}/${recipeId.id}`);
    dispatch({
      type: actionTypes.DELETE_RECIPE,
      recipeId,
    });
  };
}

export function updateRecipe(recipe) {
  return {
    type: actionTypes.UPDATE_RECIPE,
    recipe,
  };
}

export function loadRecipe(recipe) {
  return {
    type: actionTypes.LOAD_RECIPE,
    recipe,
  };
}

export function getRecipeById(recipeId) {
  console.log(`${url}/${recipeId}`);
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
