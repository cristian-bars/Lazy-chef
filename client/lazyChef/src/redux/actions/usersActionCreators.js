import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
