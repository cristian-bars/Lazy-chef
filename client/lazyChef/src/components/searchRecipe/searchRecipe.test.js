import 'react-native';
import React from 'react';
import SearchRecipe from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/recipesActionCreators';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a SearchRecipe component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;
  let route;
  beforeEach(() => {
    jest.spyOn(actions, 'loadRecipes').mockReturnValueOnce({type: ''});
    myStore = mockStore({
      recipes: [
        {
          title: 'macarrones',
          description: '1a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
        {
          title: 'Queso',
          description: '2a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
      ],
    });
    navigation = {
      navigate: jest.fn(),
    };
    route = {
      params: {
        ingredient: 'tomate',
      },
    };
  });

  test('renders correctly', () => {
    const wrapper = render(
      <Provider store={myStore}>
        <SearchRecipe route={route} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
