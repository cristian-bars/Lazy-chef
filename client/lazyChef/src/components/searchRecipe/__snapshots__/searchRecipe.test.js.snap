// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SearchRecipe component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D3E0FF",
      "justifyContent": "center",
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#397ED0",
          "borderRadius": 25,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 25,
          "height": 50,
          "margin": 15,
          "marginTop": 20,
          "paddingLeft": 15,
          "textAlign": "center",
          "width": 360,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/img/search.png",
            }
          }
          style={
            Object {
              "height": 30,
              "opacity": 0.7,
              "width": 30,
            }
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Buscar"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 20,
            "paddingLeft": 20,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "column",
            "flexWrap": "wrap",
            "margin": 10,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "white",
              "fontSize": 20,
            }
          }
        >
          Mi selecci√≥n
        </Text>
        <Text>
          tomate
        </Text>
        <Text
          style={
            Object {
              "paddingVertical": 10,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 30,
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 25,
        }
      }
    >
      Mostrar recetas
    </Text>
  </View>
</View>
`;
